PROBLEMS.md

```diff
@@ -0,0 +1,23 @@
+While IdentityCache allows to massively scale access to ActiveRecord objects, it's important to note what problems might arise with it over time and what are things that it's not solving in the current design.
+
+## Large blobs
+
+For [god objects](https://en.wikipedia.org/wiki/God_object) (like the Shop model at Shopify), it’s typical to have many associations. It’s for developers appealing to embed many of those associations into the god model’s IDC blob with `cache_has_many :images, embed: true`. As a side effect, over time the god model’s IDC blob may grow over the top. It gets even worse when some of the embedded records get larger, and the total IDC blob gets into as much as a megabyte in size. This puts pressure on both the application and memcached, because the blob needs to be serialized and unserialized each time and transferred over the network.
+
+## Overfetching
```

Comment:
Is this really a separate issue?  It seems like another issue that comes from embedded associations.

Another possible issue that can come from embedded associations is that cache invalidations for a model can be amplified by the number of models that embed them.  This can further aggravate the large cache blob problem by embedding an association that gets invalidated more frequently.